flink-tests/src/test/java/org/apache/flink/test/recovery/JobManagerHAProcessFailureRecoveryITCase.java

```diff
@@ -237,22 +217,21 @@ public void flatMap(Long value, Collector<Long> out)
                                     }
                                 });
 
-        result.output(new DiscardingOutputFormat<Long>());
+        result.output(new DiscardingOutputFormat<>());
 
         env.execute();
     }
 
-    @Test
-    public void testDispatcherProcessFailure() throws Exception {
-        final Time timeout = Time.seconds(30L);
-        final File zookeeperStoragePath = temporaryFolder.newFolder();
+    @TestTemplate
+    void testDispatcherProcessFailure() throws Exception {
```

Comment:
```suggestion
    void testDispatcherProcessFailure(@TempDir Path tempDir) throws Exception {
```
nit: you can also inject the tempDir as a parameter (instead of creating the field in the test class) since it's only used in this one test.